include/msgpack/v1/object.hpp

```diff
@@ -466,12 +466,13 @@ struct object_stringize_visitor {
         m_os << '"';
         return true;
     }
-    bool visit_bin(const char* v, uint32_t size) {
-        (m_os << '"').write(v, static_cast<std::streamsize>(size)) << '"';
+    bool visit_bin(const char* /*v*/, uint32_t size) {
+        m_os << "\"BIN(size:" << size << ")\"";
         return true;
     }
-    bool visit_ext(const char* /*v*/, uint32_t /*size*/) {
-        m_os << "EXT";
+    bool visit_ext(const char* v, uint32_t size) {
+        int type = ((size > 0) && (v[0] >= 0)) ? v[0] : -1;
+        m_os << "\"EXT(type:" << type << ",size:" << size << ")\"";
```

Comment:
See https://github.com/msgpack/msgpack/blob/master/spec.md#ext-format-family
The parameter `size` is size of type + data.
I think that users expect size of data. So the printed value should size-1.

Reply:
So the code should be

```cpp
    bool visit_ext(const char* v, uint32_t size) {
        if (size == 0) {
            m_os << "\"EXT(size:0)\"";
        }
        else {
            m_os << "\"EXT(type:" << static_cast<int>(v[0]) << ",size:" << size - 1 << ")\"";
        }
```

Reply:
Got it. Thanks. Will upload a change tomorrow morning.

Reply:
NOTE: size==0 on EXT is error. But I choose printing only size. (avoid invalid v[0] access.